src/PanelTraits/Create.php

```diff
@@ -120,4 +136,86 @@ public function syncPivot($model, $data, $form = 'create')
             }
         }
     }
+
+    private function createOneToOneRelations($item, $data, $form = 'create')
+    {
+        $relationData = $this->formatData($data, $form);
+        $this->createRelationsForItem($item, $relationData);
+    }
+
+    private function createRelationsForItem($item, $formattedData)
+    {
+        foreach ($formattedData['relations'] as $relationMethod => $relationData) {
+            $parent = $item;
+            $model = $relationData['model'];
+            $relation = $parent->{$relationMethod}();
+
+            if ($relation instanceof BelongsTo) {
+                $modelInstance = $model::find($relationData['values'])->first();
+                if ($modelInstance != null) {
+                    $relation->associate($modelInstance)->save();
+                } else {
+                    $relation->dissociate()->save();
+                }
+            } elseif ($relation instanceof HasOne) {
+                if ($parent->{$relationMethod} != null) {
+                    $parent->{$relationMethod}->update($relationData['values']);
+                    $modelInstance = $parent->{$relationMethod};
+                } else {
+                    $relationModel = new $model();
+                    $modelInstance = $relationModel->create($relationData['values']);
+                    $relation->save($modelInstance);
+                }
+            } else {
+                $relationModel = new $model();
+                $modelInstance = $relationModel->create($relationData['values']);
+                $relation->save($modelInstance);
+            }
```

Comment:
Well there... I personally would have used a ```switch``` statement, like so: https://blog.liplex.de/switch-with-instanceof-in-php/   :-)

Don't consider it important, though - just they way I would have done it :-) I'd merge it like this too, I think the code is clear enough as-is.